components/HttpsComponent/HttpsRequests.c

```diff
@@ -121,41 +87,43 @@ static void https_get_request(esp_tls_cfg_t cfg, const char *WEB_SERVER_URL, con
         }
     } while (written_bytes < strlen(REQUEST));
     ESP_LOGI(TAG, "Reading HTTP response...");
-    do
-    {
+    
+    do {
         len = sizeof(buf) - 1;
         memset(buf, 0x00, sizeof(buf));
-        // vTaskDelay((SEC*5) / portTICK_PERIOD_MS);
+
         ret = esp_tls_conn_read(tls, (char *)buf, len);
```

Comment:
needed mechanism for detecting complete response or broken response and , and if it is broken fix it with something like timeout or waiting for getting complete response ...